strongbox-commons/src/main/java/org/carlspring/strongbox/booters/SystemPropertiesInitializer.java

```diff
@@ -0,0 +1,57 @@
+package org.carlspring.strongbox.booters;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Ensures that a handful of core system properties are set prior to starting the application.
+ *
+ * @author cbono
+ */
+class SystemPropertiesInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
+{
+    @VisibleForTesting
+    static final String STRONGBOX_HOME_PROPERTY_NAME = "strongbox.home";
+
+    @VisibleForTesting
+    static final String STRONGBOX_VAULT_PROPERTY_NAME = "strongbox.vault";
+
+    @VisibleForTesting
+    static final String STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME = "strongbox.storage.booter.basedir";
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext)
+    {
+        final ConfigurableEnvironment env = applicationContext.getEnvironment();
+        final String homeDir = env.getRequiredProperty(STRONGBOX_HOME_PROPERTY_NAME);
+        final String vaultDir = env.getRequiredProperty(STRONGBOX_VAULT_PROPERTY_NAME);
+        final String storageBooterBaseDir = env.getRequiredProperty(STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME);
+
+        if (System.getProperty("logging.dir") == null)
+        {
+            System.setProperty("logging.dir", vaultDir + "/logs");
+        }
+
+        if (System.getProperty("logging.config.file") == null)
+        {
+            System.setProperty("logging.config.file", homeDir + "/etc/logback-spring.xml");
+        }
+
+        if (System.getProperty("java.io.tmpdir") == null)
+        {
+            System.setProperty("java.io.tmpdir", vaultDir + "/tmp");
+        }
+
+        if (System.getProperty("ehcache.disk.store.dir") == null)
+        {
+            System.setProperty("ehcache.disk.store.dir", homeDir + "/cache");
+        }
+
+        if (System.getProperty(STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME) == null)
+        {
+            System.setProperty(STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME, storageBooterBaseDir);
+        }
```

Comment:
Should we log a warning or info on the `else` path of this if they set the storage basedir system property but the one that is set in the application.yml differs from the one in the system property?